includes/helper/class-wp-job-manager-helper.php

```diff
@@ -207,10 +208,20 @@ public function check_for_updates( $check_for_updates_data ) {
 				&& isset( $response['new_version'] )
 				&& version_compare( $response['new_version'], $plugin_data['Version'], '>' )
 			) {
+				$available_addon_updates[ $product_slug ] = $response;
 				$check_for_updates_data->response[ $plugin_data['_filename'] ] = (object) $response;
 			}
 		}
 
+		// Enable or disable notices.
```

Comment:
I realized that it's called only when we access the Plugins or the Dashboard > Updates page. Not sure if it's called in some job too. So it just creates the notice after that.
And it will also just clear the notice only after accessing that pages when updated.

Is it something expected? Or should we implement it on a different hook?